src/video.cpp

```diff
@@ -1684,6 +1684,13 @@ enum validate_flag_e {
 
 int validate_config(std::shared_ptr<platf::display_t> &disp, const encoder_t &encoder, const config_t &config) {
   reset_display(disp, encoder.base_dev_type, config::video.output_name, config);
+
+  // User probably put in an invalid output name, let's try it again without it.
+  if(!disp && !config::video.output_name.empty()) {
+    BOOST_LOG(warning) << "Failed to open display with output name '" << config::video.output_name << "', falling back to using the next available display.";
+    reset_display(disp, encoder.base_dev_type, "", config);
+  }
+
```

Comment:
There are many areas where this could have been done, but I felt the method named validate_config() was probably the best location for this... since this issue is caused by invalid configuration.

Reply:
Actually nevermind, this is not a good spot, will need to move it inside the reset_display method. Just figuring out a way to do it without looking like bad code

Reply:
Would it make more sense to handle the error within the reset_display function?

Reply:
> Actually nevermind, this is not a good spot, will need to move it inside the reset_display method. Just figuring out a way to do it without looking like bad code

Didn't see your updated comment, ha.

Reply:
> > Actually nevermind, this is not a good spot, will need to move it inside the reset_display method. Just figuring out a way to do it without looking like bad code
> 
> Didn't see your updated comment, ha.

Just updated my commit, it's now done in the reset_display method. I changed it to a while loop, so if we ever find additional conditions we can easily extend it.

It will also now test twice with the display name removed, instead of just once.

Reply:
Of course with this change that means it's theoretically possible for it to attempt up to 4 times per encoder. 
Not sure how we would fix that scenario other than static bools though, which is something you want to tend avoid doing.